LibraBFT/Prelude.agda

```diff
@@ -352,4 +390,22 @@ module LibraBFT.Prelude where
   f-sum : ∀{a}{A : Set a} → (A → ℕ) → List A → ℕ
   f-sum f = sum ∘ List-map f
 
+  record Monad {ℓ₁ ℓ₂ : Level} (M : Set ℓ₁ → Set ℓ₂) : Set (ℓ₂ ℓ⊔ ℓ+1 ℓ₁) where
+    infixl 1 _>>=_ _>>_
+    field
+      return : ∀ {A : Set ℓ₁} → A → M A
+      _>>=_  : ∀ {A B : Set ℓ₁} → M A → (A → M B) → M B
+
+    _>>_ : ∀ {A B : Set ℓ₁} → M A → M B → M B
+    m₁ >> m₂ = m₁ >>= λ _ → m₂
+
+  open Monad ⦃ ... ⦄ public
+
+  open import Category.Monad
+  import      Data.Sum.Categorical.Left
+  instance
+    Monad-Error : ∀ {ℓ}{C : Set ℓ} → Monad{ℓ}{ℓ} (Either C)
+    Monad.return (Monad-Error{ℓ}{C}) = RawMonad.return (Data.Sum.Categorical.Left.monad C ℓ)
+    Monad._>>=_ (Monad-Error{ℓ}{C}) = RawMonad._>>=_ (Data.Sum.Categorical.Left.monad C ℓ)
```

Comment:
I'm curious about the rationale for using `Data.Sum.Categorical.Left` here.  It seems that the following suffices, which is simpler and clearer (to me, at least):
```
  instance
    Monad-Error : ∀ {ℓ}{C : Set ℓ} → Monad{ℓ}{ℓ} (Either C)
    Monad.return (Monad-Error{ℓ}{C}) = inj₂
    Monad._>>=_ (Monad-Error{ℓ}{C})  =  either (const ∘ inj₁) (flip id)
```
I was confused by the use of `Data.Sum.Categorical.Left` for `return`, which is really about `inj₂` (not ` inj₁`), but I see that `Data.Sum.Categorical.Right` works as well.  I am just not sure what benefit we expect or hope to achieve with this additional cognitive burden (again, at least for me :smile:).

Reply:
I have used the simpler definition of the instance you proposed. My idea was to show how to avoid re-implementing instances that were already defined in the standard library, but there is probably not enough variety in the implementation's use of monads to warrant the extra cognitive overhead.